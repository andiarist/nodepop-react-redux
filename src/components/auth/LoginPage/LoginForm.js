import React from 'react';
import T from 'prop-types';

import styles from './LoginForm.module.css';

import Input from '../../shared/Form/Input';
import Form from '../../shared/Form/Form';

function LoginForm({ onSubmit }) {
  return (
    <>
      <Form
        initialValues={{ email: '', password: '', remember: false }}
        onSubmit={onSubmit}>
        <p>
          <Input type="email" name="email" />
        </p>
        <p>
          <Input type="password" name="password" />
        </p>
        <p>
          <Input type="checkbox" name="remember" /> Remember
        </p>

        <button type="submit" className="button-submit">
          Log in
        </button>
      </Form>
    </>
  );
}

LoginForm.propTypes = {
  onSubmit: T.func.isRequired,
};

export default LoginForm;
