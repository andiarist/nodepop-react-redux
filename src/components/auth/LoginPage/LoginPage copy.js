import React from 'react';
import T from 'prop-types';
import { Alert, Col, Row, Typography } from 'antd';

import { login } from '../../../api/auth';
import LoginForm from './LoginForm';

const { Title } = Typography;

class LoginPage extends React.Component {
  state = {
    error: null,
  };

  handleSubmit = credentials => {
    const { onLogin, location, history, isLogged } = this.props;
    console.log('props en LoginPage:', this.props);

    console.log('onLogin en LoginPage:', onLogin);

    this.resetError();

    login(credentials)
      .then(() => {
        console.log('antes de entrar al onLogin');
        console.log('credentials:', credentials);
        //isLogged = true;
        //function onLogin(){
        //  console.log('entra en onLogin');
        //}
        onLogin(() => {
          console.log('entra en onLogin');

          // Navigate to previously required route

          const { from } = location.state || { from: { pathname: '/' } };
          console.log('from:', from);
          history.replace(from);
        });
        console.log('fuera de onLogin');
      })
      .catch(error => {
        this.setState({ error });
      });
  };

  resetError = () => this.setState({ error: null });

  render() {
    const { error } = this.state;
    return (
      <Row>
        <Col span={8} offset={8} style={{ marginTop: 64 }}>
          <Title style={{ textAlign: 'center' }}>Log In</Title>
          <LoginForm onSubmit={this.handleSubmit} />
          {error && (
            <Alert
              afterClose={this.resetError}
              closable
              message={error}
              showIcon
              type="error"
              style={{ marginTop: 24 }}
            />
          )}
        </Col>
      </Row>
    );
  }
}

LoginPage.propTypes = {
  onLogin: T.func.isRequired,
  history: T.shape({ replace: T.func.isRequired }).isRequired,
  location: T.shape({
    state: T.shape({ from: T.shape({ pathname: T.string }) }),
  }).isRequired,
  isLogged: T.bool,
};

export default LoginPage;
