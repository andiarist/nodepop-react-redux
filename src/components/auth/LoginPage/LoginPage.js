import React from 'react';
import { connect } from 'react-redux';
import T from 'prop-types';
import { Alert, Col, Row, Typography } from 'antd';

//import { login } from '../../../api/auth';
import LoginForm from './LoginForm';

//import {
//  authLoginRequest,
//  authLoginSuccess,
//  authLoginFailure,
//} from '../../../store/actions';
import { login } from '../../../store/actions';
import { getUi } from '../../../store/selectors';

const { Title } = Typography;

function LoginPage({ onLogin, location, history, error, loading }) {
  const handleSubmit = credentials => {
    onLogin(credentials, location, history);
  };

  return (
    <Row>
      <Col span={8} offset={8} style={{ marginTop: 64 }}>
        <Title style={{ textAlign: 'center' }}>Log In</Title>
        <LoginForm onSubmit={handleSubmit} />
        {error && (
          <Alert
            closable
            message={error}
            showIcon
            type="error"
            style={{ marginTop: 24 }}
          />
        )}
      </Col>
    </Row>
  );
}

LoginPage.propTypes = {
  onLogin: T.func,
  history: T.shape({ replace: T.func.isRequired }).isRequired,
  location: T.shape({
    state: T.shape({ from: T.shape({ pathname: T.string }) }),
  }).isRequired,
};

const mapStateToProps = state => getUi(state);

const mapDispatchToProps = dispatch => ({
  onLogin: (credentials, location, history) =>
    dispatch(login(credentials, location, history)),
});

//export default connect(getUi, login)(LoginPage);
export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
