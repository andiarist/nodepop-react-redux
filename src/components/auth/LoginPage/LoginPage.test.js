import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import LoginPage from './LoginPage';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};
const store = mockStore(initialState);

describe('Testing <LoginPage />', () => {
  const props = {
    onLogin: jest.fn(),
    history: {},
    location: {},
    error: null,
    loading: false,
  };
  const render = () =>
    shallow(
      <Provider store={store}>
        <LoginPage {...props} />
      </Provider>,
    );
  test('snapshot', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
});
