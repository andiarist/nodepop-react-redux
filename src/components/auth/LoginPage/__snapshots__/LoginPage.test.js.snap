// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LoginPage /> snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(LoginPage)
    error={null}
    history={Object {}}
    loading={false}
    location={Object {}}
    onLogin={[MockFunction]}
  />
</ContextProvider>
`;
